deepmd/descriptor/se_a.py

```diff
@@ -736,7 +736,7 @@ def _filter_lower(
               if (not self.uniform_seed) and (self.seed is not None): self.seed += self.seed_shift
           else:
             # we can safely return the final xyz_scatter filled with zero directly
-            return tf.cast(tf.fill((natom, 4, outputs_size[-1]), 0.), GLOBAL_TF_FLOAT_PRECISION)
+            return tf.cast(tf.fill((natom, 4, outputs_size[-1]), 0.), self.filter_precision)
```

Comment:
We cast the result back to global precision before return

Reply:
I don't see any `cast` in line 722 or line 747

Reply:
Is there something wrong?

Reply:
> I don't see any `cast` in line 722 or line 747

747 should be cast back...

Why do we need to cast at L722?

Reply:
Instead of `_filter_lower`, I think we should cast back at line 839 before `_filter` returns.

Reply:
I agree with you

Reply:
Why not writing a decorator for doing that? It casts the inputs of `_filter` to `filter_precision` and casts back to global precision when `_filter` returns

Reply:
Good idea!